transformers/modeling_bert.py

```diff
@@ -329,12 +432,35 @@ def forward(self, hidden_states, attention_mask=None, head_mask=None):
         return outputs
 
 
+class BertDecoderLayer(nn.Module):
+    def __init__(self, config):
+        super(BertDecoderLayer, self).__init__()
+        self.self_attention = BertAttention(config)
+        self.attention = BertDecoderAttention(config)
+        self.intermediate = BertIntermediate(config)
+        self.output = BertOutput(config)
+
+    def forward(self, hidden_states, encoder_outputs, attention_mask=None, head_mask=None):
+        self_attention_outputs = self.self_attention(hidden_states, attention_mask, head_mask)
+        self_attention_output = self_attention_outputs[0]
+        attention_outputs = self.attention(query=self_attention_output,
+                                           key=encoder_outputs,
+                                           value=encoder_outputs,
+                                           attention_mask=attention_mask,
+                                           head_mask=head_mask)
+        attention_output = attention_outputs[0]
+        intermediate_output = self.intermediate(attention_output)
+        layer_output = self.output(intermediate_output, attention_output)
+        outputs = (layer_output,) + attention_outputs[1:]
```

Comment:
Same here, let's try to use a single class and take care of calling the additional encoder-decoder attention somewhere like here if a specific flag is set in the configuration.